AlphaWallet/TokenScriptClient/Coordinators/EventSourceCoordinator.swift

```diff
@@ -46,24 +46,31 @@ final class EventSourceCoordinator: NSObject {
     private func setupWatchingTokenScriptFileChangesToFetchEvents() {
         //TODO this is firing twice for each contract. We can be more efficient
         assetDefinitionStore.bodyChange
-            .receive(on: RunLoop.main)
+            .receive(on: queue)
             .compactMap { [weak self] in self?.tokensDataStore.token(forContract: $0) }
```

Comment:
Is the assumption is that `tokensDatastore.store(forContract:)` (and everything else) will always access Realm in the main thread?

Reply:
actually nope, its able to access to it from any thread, when we access to it from main thread it just doesn't move to sync queue,  and preforms as is. but whe we access to this part from different from main thread it blocks on sync queue, and performs code sync.